src/app/shared/entry/doi/manage-dois/manage-dois-dialog.component.ts

```diff
@@ -90,6 +101,21 @@ export class ManageDoisDialogComponent {
     this.manageDoisDialogService.saveDoiSelection(this.entry, this.selectedOption);
   }
 
+  requestDoiAccessLink() {
+    this.manageDoisDialogService.requestDoiAccessLink(this.entry).subscribe((accessLink) => {
+      this.dockstoreDoiAccessLink = accessLink;
+    });
+  }
+
+  deleteDoiAccessLink() {
+    this.manageDoisDialogService.deleteDoiAccessLink(this.entry);
+    this.dockstoreDoiAccessLink = undefined;
```

Comment:
Borderline nitpicky, but we might not unset `dockstoreDoiAccessLink` until the observable has successfully completed.  That way, the user can try again in the case of failure (due to network problems or whatever).  With this change, the implementation/flow of `deleteDoiAccessLink` looks more similar to `requestDoiAccessLink`.